File: 940-fruit-into-baskets/fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        int l=0,r=0,maxFruits=0;
        unordered_map<int,int>fruitCount;

        while(r<fruits.size()){
            fruitCount[fruits[r]]++;

            if(fruitCount.size()>2){
                fruitCount[fruits[l]]--;
                if(fruitCount[fruits[l]]==0){
                    fruitCount.erase(fruits[l]);
                }
                l++;
            }
            if(fruitCount.size()<=2){
                maxFruits = max(maxFruits,r-l+1);
            }
            r++;
        }
        return maxFruits;
    }
};

File: 940-fruit-into-baskets/fruit-into-baskets.go
func totalFruit(fruits []int) int {

    fruitMap := make(map[int]int)
    l:=0
    r:=0
    maxFruits :=0

    for r< len(fruits){
        fruitMap[fruits[r]]++

        if len(fruitMap) >2{
            fruitMap[fruits[l]]--
            if  fruitMap[fruits[l]] ==0{
                delete(fruitMap, fruits[l])
            }
            l++
        }
        if len(fruitMap) <=2{
            maxFruits = max(maxFruits, r-l+1)
        }
        r++
    }
    return maxFruits
}